### ISSUE TITLE
Window Focus falls back to parent on system file dialog when interacting with mouse

### ISSUE THREAD

**Issue by BrunoWallner** [pop-os/cosmic-comp#455]
## Description
When using something like [rfd](https://github.com/PolyMeilex/rfd) to open a file dialog window, the focus properly switches to the new window. But while interacting with it with the mouse, the focus will switch back to its parent and it is impossible to click anything in the file dialog.
A workaround is to use `super` + `←↓↑→` to focus it again and then use keyboard inputs only to interact with the child.

## How to reproduce
1. clone `https://codeberg.org/BrunoWallner/apoxel` with commit ID: `e331ddc66f4a0613c25ea38c2bbecc4a170f711c`
2. cd into `path`
3. run: `cargo r --release`
4. on the `Load world` menu select `vox importer` and click on the `Open file` button
5. use the mouse to interact with window that just opened up

## System information
- Linux version: `6.8.7-arch1-2`
- cosmic-comp commit ID: `5e61ce1ff8606986e24edb6a91c0eb9a175c305f`
- cosmic-epoch DE installed via AUR with commit ID: `2eadc4eca2bc2f8ecf6ecbe0ebe61de08544686c`

**Comment by ryanabx** [comment]
Just to clarify - you can interact with the focused window keyboard inputs, but not mouse inputs?


If so that'd be similar to an issue I have with visual studio code where sometimes a window will completely lose the ability to receive any kind of mouse event, but keyboard events are fine

**Comment by BrunoWallner** [comment]
> Just to clarify - you can interact with the focused window keyboard inputs, but not mouse inputs?
> 
> If so that'd be similar to an issue I have with visual studio code where sometimes a window will completely lose the ability to receive any kind of mouse event, but keyboard events are fine

Yes exactly that is the case. At first it is properly focused and I can use keyboard inputs, but after interacting with the mouse, the focus is lost and I cannot either click on it or use the keyboard.

**Comment by spgarbet** [comment]
I think this is related:

If I have an application within a defined square, sometimes a popup menu will extend outside of the parent window square. If I click with my mouse on a menu item outside the parent square that click event goes to the application underneath the popup and not the application that generated the popup. This happens across multiple applications. A good example application is RStudio and just use the copy/paste menu popup.

I opened it in the wrong project. https://github.com/pop-os/cosmic-greeter/issues/250#issuecomment-2945133642

**Comment by tcamargo** [comment]
I have a similar issue with a long dropdown in Orcaslicer.

<img width="470" height="570" alt="Image" src="https://github.com/user-attachments/assets/f072a33b-d741-4ed0-8bd0-05b1a0b733bf" />

I can´t select eSUN PLA+. Focus/click goes to he underneath window.

Using COSMIC Alpha 7.

**Comment by KodeBarista** [comment]
Has it been fixed? Orcalslicer drop-downs seemed work as expected for me. Can you verify?

**Comment by spgarbet** [comment]
For me the menu item fires correctly, but the focus still changes. This is a lot better and far more workable. My test was using gvim over firefox. Personally I think this should be closed and a new issue opened on the focus change. It's related but treatable as a separate issue.
